components/elm/src/main/pftvarcon.F90

```diff
@@ -1103,12 +1223,209 @@ subroutine pftconrd
+             call endrun(msg=' ERROR: when create_crop_landunit is true, last crop must be the last one of all PFTs .'//errMsg(__FILE__, __LINE__))
+          end if
+
+          natpft_size = (numpft + 1) - numcft    ! note that numpft doesn't include bare ground -- thus we add 1
+          cft_size    = numcft
+       else
+          natpft_size = numpft + 1               ! note that numpft doesn't include bare ground -- thus we add 1
+          cft_size    = 0
+       end if
+       natpft_lb = 0
+       natpft_ub = natpft_lb + natpft_size - 1
+       cft_lb = natpft_ub + 1
+       cft_ub = max(cft_lb, cft_lb + cft_size - 1)            ! NOTE: if cft_size is ZERO, could be issue (but so far so good)
+       surfpft_lb  = natpft_lb
+       surfpft_ub  = natpft_ub
+       surfpft_size = natpft_size
+       max_patch_per_col= max(numpft+1, numcft, maxpatch_urb)
+
+
+   end if  ! end if 'PFT_DEFAULT'
+
+     ! checking of pft flags' conflict
+     if ( .not. use_fates ) then
+        do i = 0, mxpft
+          if (i == noveg) then
+             if ( (nonvascular(i)+woody(i)+graminoid(i)+max(iscft(i),crop(i)+percrop(i))) >= 1 .or. &
+                  (needleleaf(i)+evergreen(i)+stress_decid(i)+season_decid(i)+nfixer(i)) >= 1 ) then
+                print *, 'ERROR: Incorrect not-vegetated PFT flags: ', i, ' ', trim(pftname(i))
+                call endrun(msg=' ERROR: not_vegetated has at least one positive PFT flag '//errMsg(__FILE__, __LINE__))
+             end if
+
+          else if ( (nonvascular(i)+woody(i)+graminoid(i)+max(iscft(i),crop(i)+percrop(i))) >= 1) then
+             if (nonvascular(i) >= 1 .and. (woody(i)+graminoid(i)+max(iscft(i),crop(i)+percrop(i))) >= 1) then
+                print *, 'ERROR: Incorrect nonvasculr PFT flags: ', i, ' ', trim(pftname(i))
+                call endrun(msg=' ERROR: nonvascular PFT cannot be any of woody/graminoid/crop type '//errMsg(__FILE__, __LINE__))
+             else if (woody(i) >= 1 .and. (nonvascular(i)+graminoid(i)+max(iscft(i),crop(i)+percrop(i))) >= 1) then
+                print *, 'ERROR: Incorrect woody PFT flags: ', i, ' ', trim(pftname(i))
+                call endrun(msg=' ERROR: woody PFT cannot be any of nonvascular/graminoid/crop type - '//errMsg(__FILE__, __LINE__))
+             else if (graminoid(i) >= 1 .and. (nonvascular(i)+woody(i)+max(iscft(i),crop(i)+percrop(i))) >=1 ) then
+                print *, 'ERROR: Incorrect graminoid PFT flags: ', i, ' ', trim(pftname(i))
+                call endrun(msg=' ERROR: graminoid PFT cannot be any of nonvascular/woody/crop type - '//errMsg(__FILE__, __LINE__))
+             else if ( (max(iscft(i),crop(i)+percrop(i))) >= 1 .and. (nonvascular(i)+woody(i)+graminoid(i)) >= 1) then
+                print *, 'ERROR: Incorrect crop PFT flags: ', i, ' ', trim(pftname(i))
+                call endrun(msg=' ERROR: crop PFT cannot be any of nonvascular/woody/graminoid type - '//errMsg(__FILE__, __LINE__))
+             end if
```

Comment:
We have some woody crops targeted for future versions of the model, so we might want more flexibility with the flags.

Reply:
These error checks are very specific for the current usage of woody etc. I agree something more flexible would be good. Let's work on that together as a future PR.